wagtail/admin/views/chooser.py

```diff
@@ -129,10 +130,54 @@ def browse(request, parent_page_id=None):
     parent_page.can_choose = can_choose_page(
         parent_page, permission_proxy, desired_classes, can_choose_root, user_perm, target_pages=target_pages, match_subclass=match_subclass)
 
+    selected_locale = None
+    locale_options = []
     show_locale_labels = getattr(settings, 'WAGTAIL_I18N_ENABLED', False)
     if show_locale_labels:
         pages = pages.select_related('locale')
 
+        # 'locale' is the current value of the "Locale" selector in the UI
+        if request.GET.get('locale'):
+            selected_locale = get_object_or_404(Locale, language_code=request.GET['locale'])
+            active_locale_id = selected_locale.pk
+        else:
+            active_locale_id = Locale.get_active().pk
+
+        if parent_page_id is None:
+            # we are at the Root level, so get the locales from the current pages
+            choose_url = reverse('wagtailadmin_choose_page')
+            locale_options = [
+                {
+                    'locale': locale,
+                    'url': choose_url + '?' + urlencode({
+                        'page_type': page_type_string,
+                        'locale': locale.language_code
+                    })
+                }
+                for locale in Locale.objects.filter(pk__in=pages.values_list('locale_id')).exclude(pk=active_locale_id)
+            ]
+        else:
+            # we have a parent page (that is not the root page), so get the locales based on the available translations
```

Comment:
If we have a parent page, should we set ``selected_locale`` so the locale of the current parent is shown in the dropdown?

Reply:
sorted in b8c35c5